### PR TITLE
refactor: direct use 'client.Object' instead of 'meta.Accessor()'

### PR DESCRIPTION
**Pull request by xcoulon**
see also https://github.com/codeready-toolchain/toolchain-common/pull/210

Fixes https://issues.redhat.com/browse/CRT-1182

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: test/fake/masteruserrecord_client.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	crtapi "github.com/codeready-toolchain/api/api/v1alpha1"
 
 	"github.com/stretchr/testify/require"
-	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (alexeykazakov) [inline test/fake/masteruserrecord_client.go:11]: Duplicates

File: test/fake/signup_client.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 
 	"github.com/gofrs/uuid"
 	"github.com/stretchr/testify/require"
-	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (alexeykazakov) [inline test/fake/signup_client.go:14]: And here

File: go.mod [hunk 3]
```diff
@@ -34,4 +34,6 @@ require (
 	sigs.k8s.io/controller-runtime v0.8.3
 )
 
+replace github.com/codeready-toolchain/toolchain-common => github.com/xcoulon/toolchain-common v0.0.0-20210914134654-45c375935ced
```
Comment (alexeykazakov) [inline go.mod:37]: What do you need this for?

Comment (xcoulon) [reply go.mod:37]: there are similar changes in https://github.com/codeready-toolchain/toolchain-common/pull/210 and I think we should build/test on those, too.

Comment (xcoulon) [reply go.mod:37]: I updated the description to mention this dependency

File: test/fake/masteruserrecord_client.go (hunk 1 above)
Comment (xcoulon) [reply test/fake/masteruserrecord_client.go]: fixed in f3e2091

File: test/fake/signup_client.go (hunk 2 above)
Comment (xcoulon) [reply test/fake/signup_client.go]: fixed in f3e2091

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**